libs/header/sync/metrics.go

```diff
@@ -0,0 +1,47 @@
+package sync
+
+import (
+	"context"
+	"sync/atomic"
+
+	"go.opentelemetry.io/otel/metric/global"
+	"go.opentelemetry.io/otel/metric/instrument"
+)
+
+var meter = global.MeterProvider().Meter("header/sync")
```

Comment:
since we now have two `header` folders, I'd suggest renaming it to `libs/syncer`.
 @renaynay , wdyt?

Reply:
might be hard to differentiate for metrics-sake

Reply:
keeping it as is ftm unless you guys think otherwise